catalog/security/catalog-security-resourceplugin/src/main/java/org/codice/ddf/catalog/security/ResourceUriPolicy.java

```diff
@@ -0,0 +1,187 @@
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.security;
+
+import java.io.Serializable;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.opengis.filter.Filter;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.Query;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.plugin.PolicyPlugin;
+import ddf.catalog.plugin.PolicyResponse;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.catalog.plugin.impl.PolicyResponseImpl;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.security.permission.Permissions;
+
+/**
+ * Restricts how resource URIs are updated and created.
+ * There are security risk to allowing users to update URIs as well as allowing users to
+ * pass in a URI when the metacard is created.
+ */
+public class ResourceUriPolicy implements PolicyPlugin {
+
+    private String[] createPermissions;
+
+    private String[] updatePermissions;
+
+    private CatalogFramework catalogFramework;
+
+    private FilterBuilder filterBuilder;
+
+    @Override
+    public PolicyResponse processPreCreate(Metacard input, Map<String, Serializable> properties)
+            throws StopProcessingException {
+
+        if (input.getResourceURI() != null && StringUtils.isNotEmpty(input.getResourceURI()
+                .toString())) {
+            return new PolicyResponseImpl(null,
+                    Permissions.parsePermissionsFromString(getCreatePermissions()));
+        }
+
+        return new PolicyResponseImpl();
+    }
+
+    @Override
+    public PolicyResponse processPreUpdate(Metacard input, Map<String, Serializable> properties)
+            throws StopProcessingException {
+
+        if (requiresPermissionsToChangeFrom(input.getResourceURI(),
+                getMetacardFromCatalog(input.getId()).getResourceURI())) {
+            return new PolicyResponseImpl(null,
+                    Permissions.parsePermissionsFromString(getUpdatePermissions()));
+        } else {
+            return new PolicyResponseImpl();
+        }
+    }
+
+    private boolean requiresPermissionsToChangeFrom(URI input, URI catalog) {
+
+        return !uriToString(input).equals(uriToString(catalog));
+    }
+
+    private String uriToString(URI uri) {
+        return uri == null ? "" : uri.toString();
```

Comment:
Are `null` and `""` really equivalent? If the `resource-uri` is null on the update metacard, won't that have no effect on the existing `resource-uri`?